packages/jest-config/src/normalize.js

```diff
@@ -319,6 +319,11 @@ function normalize(config: InitialConfig, argv: Object = {}) {
       case 'unmockedModulePathPatterns':
         value = normalizeUnmockedModulePathPatterns(config, key);
         break;
+
+      case 'testRegex':
+        value = config[key];
+        break;
```

Comment:
Gah, that's a leftover from an earlier implementation, when the deprecation warnings were generated here. Good catch.